src/analysis/schwartz_space.lean

```diff
@@ -90,6 +90,18 @@ begin
   exact ⟨max C 1, by positivity, λ x, (hC x).trans (le_max_left _ _)⟩,
 end
 
+lemma is_O_cocompact (f : 𝓢(E, F)) (k : ℕ) :
+  asymptotics.is_O (filter.cocompact E) f (λ x, ‖x‖ ^ (-k : ℤ)) :=
```

Comment:
Can you prove the more general fact that `f` is a big O of `(λ x, ‖x‖ ^ k)` for any `k : ℤ`, which contains strictly this one? And also prove it for real `k`? (So that you won't need casts when you use it in Poisson summation).

Reply:
I've added the real case (but without passing through the general integer case, I hope that's OK).

Reply:
I would still be happy to have the general integer case, since it is a strict generalization of `is_O_cocompact_zpow_neg_nat`. But now you can deduce it easily from the real case!

Also, I would move these lemmas a few lines below (after the `ext` statement) because they are less fundamental.